src/dataflow-types/src/types.rs

```diff
@@ -1387,6 +1388,31 @@ pub mod sources {
             Header { names: Vec<String> },
         }
 
+        impl RustType<ProtoColumnSpec> for ColumnSpec {
+            fn into_proto(self: &Self) -> ProtoColumnSpec {
+                use proto_column_spec::{Kind, ProtoHeader};
+                ProtoColumnSpec {
+                    kind: Some(match self {
+                        ColumnSpec::Count(c) => Kind::Count(c.into_proto()),
+                        ColumnSpec::Header { names } => Kind::Header(ProtoHeader {
+                            names: names.clone(),
+                        }),
+                    }),
+                }
+            }
+
+            fn from_proto(proto: ProtoColumnSpec) -> Result<Self, TryFromProtoError> {
+                use proto_column_spec::{Kind, ProtoHeader};
+                let kind = proto
+                    .kind
+                    .ok_or_else(|| TryFromProtoError::missing_field("ProtoColumnSpec::kind"))?;
```

Comment:
`into_rust_if_some` and/or `try_into_if_some` are basically doing it for types that have a non-Proto equivalent, but since `Kind` is just an intermediate type generated by Prost that doesn't map cleanly to the enum, it doesn't work.

(and my Rust-fu is not good enough to think about another way of doing it)